guava/src/main/java/com/fasterxml/jackson/datatype/guava/ser/RangeSerializer.java

```diff
@@ -109,10 +130,28 @@ public void serializeWithType(Range<?> value, JsonGenerator gen, SerializerProvi
         throws IOException
     {
         gen.setCurrentValue(value);
-        WritableTypeId typeIdDef = typeSer.writeTypePrefix(gen,
-                typeSer.typeId(value, JsonToken.START_OBJECT));
-        _writeContents(value, gen, provider);
-        typeSer.writeTypeSuffix(gen, typeIdDef);
+        if (shape == JsonFormat.Shape.STRING){
```

Comment:
This looks correct. But made me realize that it may be tricky to support polymorphic handling for deserialization when value shape may vary (Scalar and Object values support different types of Type Id inclusion).